### PR TITLE
Add ghost data routines to advection subcell

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

Classes for sending out ghost data for FD reconstruction.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/ScalarAdvection/Subcell/GhostData.hpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
```
Comment (nilsdeppe) [inline src/Evolution/Systems/ScalarAdvection/Subcell/GhostData.hpp]: you never need `GenerateInstantiations.hpp` in a header file

File: src/Evolution/Systems/ScalarAdvection/Subcell/GhostData.hpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <typename T>
+class Variables;
+namespace Tags {
+template <typename TagsList>
+struct Variables;
+}  // namespace Tags
```
Comment (nilsdeppe) [inline src/Evolution/Systems/ScalarAdvection/Subcell/GhostData.hpp:21]: need forward decl of `Mesh`

File: src/Evolution/Systems/ScalarAdvection/Subcell/GhostData.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#include "Evolution/Systems/ScalarAdvection/Subcell/GhostData.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/Variables.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Projection.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
```
Comment (nilsdeppe) [inline src/Evolution/Systems/ScalarAdvection/Subcell/GhostData.cpp:14]: include `NumericalAlgorithms/Spectral/Mesh.hpp`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
You can squash immediately

**Comment by yoonso0-0** [comment]
Fixed!

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**